x/gamm/keeper/internal/events/emit.go

```diff
@@ -0,0 +1,70 @@
+package events
+
+import (
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+)
+
```

Comment:
Unless we have multiple places where we call `Emit*Event`, I'd say extracting these into functions is overkill IMO.

Reply:
There are indeed multiple places where events are emitted. In addition, the events differ as well as their attributes.

Having each event encapsulated in a function allows for asserting the kind of attributes emitted by events in tests. Otherwise, they would be left completely untested

An example where attributes are tested:
https://github.com/osmosis-labs/osmosis/blob/ad633616412c32d8918411e06120b65a9fd84a07/x/gamm/keeper/internal/events/emit_test.go#L61-L70